src/chrome/internal/sources/loadedSource.ts

```diff
@@ -0,0 +1,101 @@
+import { IScript } from '../scripts/script';
+import { CDTPScriptUrl } from './resourceIdentifierSubtypes';
+import { IResourceIdentifier, parseResourceIdentifier, ResourceName } from './resourceIdentifier';
+import { determineOrderingOfStrings } from '../../collections/utilities';
+
+/** This interface represents a source or text that is related to a script that the debugee is executing. The text can be the contents of the script itself,
+ *  or a file from which the script was loaded, or a file that was compiled to generate the contents of the script
+ */
+export interface ILoadedSource<TString = string> {
+    readonly script: IScript;
+    readonly identifier: IResourceIdentifier<TString>;
+    readonly origin: string;
```

Comment:
What does origin refer to? Strange to see just a bare string type in here 😄

Reply:
This property serves the purpose of the origin property in the VS Code protocol: https://github.com/Microsoft/vscode-debugadapter-node/blob/64665fe18a0316b2a871df4b36cff214c28d2598/protocol/src/debugProtocol.ts#L1315

We'll use it to explain where we got the sources from (memory, source map, disk, etc...)